Src/Plugins/Security/SecurePlaintextSecretsValidators/AwsHttpRequestSigner.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Cryptography;
+using System.Text;
+
+using Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk;
+
+namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Plugins.Security
+{
+    // https://docs.aws.amazon.com/general/latest/gr/create-signed-request.html
+
+    internal class AwsHttpRequestSigner : IDisposable
+    {
+        private const string ISO8601BasicDateTimeFormat = "yyyyMMddTHHmmssZ";
+
+        private readonly SHA256 sha256;
+        private readonly string accessKey;
+        private readonly string accessSecret;
+
+        public AwsHttpRequestSigner(string accessKey, string accessSecret)
+        {
+            this.accessKey = accessKey;
+            this.accessSecret = accessSecret;
+            this.sha256 = SHA256.Create();
+        }
+
+        public virtual string Version => "AWS4";
```

Comment:
>Version [](http://example.com/codeflow?start=30&length=7)

these virtual members can be overridden by derived class with new values e.g. Google Cloud storage use "GOOG4" etc.